account_journal_lock_date/tests/test_journal_lock_date.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import date, timedelta
+
+from odoo.exceptions import UserError
+from odoo.tests import tagged
+
+from odoo.addons.account.tests import common
+
+
+@tagged("post_install", "-at_install")
+class TestJournalLockDate(common.AccountTestInvoicingCommon):
+    def setUp(self):
+        super(TestJournalLockDate, self).setUp()
+        self.account_move_obj = self.env["account.move"]
+        self.account_move_line_obj = self.env["account.move.line"]
+        self.company_id = self.ref("base.main_company")
+        self.partner = self.browse_ref("base.res_partner_12")
+
+        self.account = self.company_data["default_account_revenue"]
+        self.account2 = self.company_data["default_account_expense"]
+        self.journal = self.company_data["default_journal_bank"]
+
+        # create a move and post it
+        self.move = self.account_move_obj.create(
+            {
+                "date": date.today(),
+                "journal_id": self.journal.id,
+                "line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "account_id": self.account.id,
+                            "credit": 1000.0,
+                            "name": "Credit line",
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "account_id": self.account2.id,
+                            "debit": 1000.0,
+                            "name": "Debit line",
+                        },
+                    ),
+                ],
+            }
+        )
+        self.move.action_post()
+        # lock journal, set 'Lock Date for Non-Advisers'
+        self.journal.period_lock_date = date.today() + timedelta(days=2)
+
+    def test_journal_lock_date(self):
+        self.env.user.write({"groups_id": [(3, self.ref("base.group_system"))]})
+        self.env.user.write(
+            {"groups_id": [(3, self.ref("account.group_account_manager"))]}
+        )
+        self.assertFalse(self.env.user.has_group("account.group_account_manager"))
+
+        # Test that the move cannot be written, or cancelled
+        with self.assertRaises(UserError):
```

Comment:
@RodrigoBM do you mind to replace all the occurrences of `with self.assertRaises(UserError):` with:

```suggestion
        with self.assertRaisesRegex(UserError, ".*prior to and inclusive of the lock date.*"):
```

This way we ensure that the correct error is raised.

Reply:
done